compiler/src/iree/compiler/GlobalOptimization/test/materialize_homogeneous_encodings.mlir

```diff
@@ -0,0 +1,97 @@
+#device_target_llvm_cpu = #hal.device.target<"llvm-cpu", {executable_targets = [#executable_target_embedded_elf_x86_64_]}>
+module attributes {hal.device.targets = [#device_target_llvm_cpu]} {
+  func.func @matmul_1025x513x257(%arg0: !hal.buffer_view) -> !hal.buffer_view attributes {iree.abi.stub} {
+    %cst = arith.constant 0.000000e+00 : f32
+    %cst_0 = arith.constant dense<2.000000e+00> : tensor<257x513xf32>
+    %0 = hal.tensor.import %arg0 "input 0" : !hal.buffer_view -> tensor<1025x257xf32>
+    %1:2 = iree_linalg_ext.upper_bound_tile_size tensor<1025x257xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = LHS>> -> index, index
+    %2 = affine.apply #map()[%1#0]
+    %3 = affine.apply #map1()[%1#1]
+    %padded = tensor.pad %0 low[0, 0] high[%2, %3] {
+    ^bb0(%arg1: index, %arg2: index):
+      tensor.yield %cst : f32
+    } : tensor<1025x257xf32> to tensor<?x?xf32>
+    %4 = iree_linalg_ext.set_encoding %padded : tensor<?x?xf32> -> tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = LHS, original_type = tensor<1025x257xf32>>>
+    %5:2 = iree_linalg_ext.upper_bound_tile_size tensor<257x513xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RHS>> -> index, index
+    %6 = affine.apply #map1()[%5#0]
+    %7 = affine.apply #map2()[%5#1]
+    %padded_1 = tensor.pad %cst_0 low[0, 0] high[%6, %7] {
+    ^bb0(%arg1: index, %arg2: index):
+      tensor.yield %cst : f32
+    } : tensor<257x513xf32> to tensor<?x?xf32>
+    %8 = iree_linalg_ext.set_encoding %padded_1 : tensor<?x?xf32> -> tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RHS, original_type = tensor<257x513xf32>>>
+    %9:2 = iree_linalg_ext.upper_bound_tile_size tensor<1025x513xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT>> -> index, index
+    %10 = affine.apply #map3()[%9#0]
+    %11 = affine.apply #map4()[%9#1]
+    %12 = tensor.empty(%10, %11) : tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT, original_type = tensor<1025x513xf32>>>
+    %13 = linalg.fill ins(%cst : f32) outs(%12 : tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT, original_type = tensor<1025x513xf32>>>) -> tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT, original_type = tensor<1025x513xf32>>>
+    %14 = linalg.matmul ins(%4, %8 : tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = LHS, original_type = tensor<1025x257xf32>>>, tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RHS, original_type = tensor<257x513xf32>>>) outs(%13 : tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT, original_type = tensor<1025x513xf32>>>) -> tensor<?x?xf32, #iree_linalg_ext.encoding<user = MATMUL_F32F32F32, role = RESULT, original_type = tensor<1025x513xf32>>>
```

Comment:
this is a large test - are we sure we need all this IR and non-local affine maps and such? these become change detectors and a burden to maintain as there's no real way to make incremental changes without recapturing them from something else. Could this instead just be one ---- per op and ensure the ops get converted instead of trying to capture an entire real dispatch?